spec/unit/controllers/internal/syslog_drain_urls_controller_spec.rb

```diff
@@ -279,50 +279,100 @@ module VCAP::CloudController
       let(:app_obj) { AppModel.make(name: 'app-1', space: space) }
       let(:app_obj2) { AppModel.make(name: 'app-2', space: space) }
       let(:app_obj3) { AppModel.make(name: 'app-3', space: space) }
+      let(:app_obj4) { AppModel.make(name: 'app-4', space: space) }
       let(:instance1) { UserProvidedServiceInstance.make(space: app_obj.space) }
       let(:instance2) { UserProvidedServiceInstance.make(space: app_obj.space) }
       let(:instance3) { UserProvidedServiceInstance.make(space: app_obj2.space) }
       let(:instance4) { UserProvidedServiceInstance.make(space: app_obj3.space) }
+      let(:instance5) { UserProvidedServiceInstance.make(space: app_obj3.space) }
+      let(:instance6) { UserProvidedServiceInstance.make(space: app_obj4.space) }
+      let(:instance7) { UserProvidedServiceInstance.make(space: app_obj.space) }
+      let(:instance8) { UserProvidedServiceInstance.make(space: app_obj2.space) }
       let!(:binding_with_drain1) { ServiceBinding.make(syslog_drain_url: 'fish,finger', app: app_obj, service_instance: instance1) }
       let!(:binding_with_drain2) { ServiceBinding.make(syslog_drain_url: 'foobar', app: app_obj, service_instance: instance2) }
       let!(:binding_with_drain3) { ServiceBinding.make(syslog_drain_url: 'foobar', app: app_obj2, service_instance: instance3) }
       let!(:binding_with_drain4) { ServiceBinding.make(
         syslog_drain_url: 'barfoo',
         app: app_obj3,
         service_instance: instance4,
-        credentials: { 'cert' => 'a cert', 'key' => 'a key' })
+        credentials: { 'cert' => 'cert1', 'key' => 'key1' })
+      }
+      let!(:binding_with_drain5) { ServiceBinding.make(
+        syslog_drain_url: 'barfoo2',
+        app: app_obj,
+        service_instance: instance7,
+        credentials: { 'cert' => 'cert1', 'key' => 'key1' })
+      }
+      let!(:binding_with_drain6) { ServiceBinding.make(
+        syslog_drain_url: 'barfoo2',
+        app: app_obj2,
+        service_instance: instance8,
+        credentials: { 'cert' => 'cert1', 'key' => 'key1' })
+      }
+      let!(:binding_with_drain7) { ServiceBinding.make(
+        syslog_drain_url: 'barfoo2',
+        app: app_obj3,
+        service_instance: instance5,
+        credentials: { 'cert' => 'cert2', 'key' => 'key2' })
+      }
+      let!(:binding_with_drain8) { ServiceBinding.make(
+        syslog_drain_url: 'barfoo2',
+        app: app_obj4,
+        service_instance: instance6,
+        credentials: { 'cert' => 'cert2', 'key' => 'key2' })
       }
 
       it 'returns a list of syslog drain urls and their credentials' do
         get '/internal/v5/syslog_drain_urls', '{}'
         expect(last_response).to be_successful
 
-        expect(decoded_results.count).to eq(3)
+        expect(decoded_results.count).to eq(4)
 
         foobar = decoded_results.select { |result| result['url'] == 'foobar' }.first
```

Comment:
My proposal is to first sort the complete results and then have a single `expect().to eq`:
```
sorted_results = decoded_results.sort { |a, b| a['url'] <=> b['url'] }.each do |binding|
  binding['credentials'].sort! { |a, b| a['cert'] <=> b['cert'] }.each do |credential|
    credential['apps'].sort! { |a, b| a['hostname'] <=> b['hostname'] }
  end
end
```

And then:
```
expect(sorted_results).to eq([
  ... the expected (sorted) result ...
])
```